server/services/user.js

```diff
@@ -1,16 +1,18 @@
 const { logger } = require('./logger')
 
-const userServiceFactory = elite2Api => {
-  async function setActiveCaseloadToTheDefaultOne(context, details) {
-    const caseLoads = await elite2Api.get(context, '/api/users/me/caseLoads')
-    if (caseLoads.length > 0) {
-      const firstCaseLoad = caseLoads[0].caseLoadId
-      logger.warn({ details }, `No active caseload set: setting to ${firstCaseLoad}`)
-      await elite2Api.put(context, '/api/users/me/activeCaseLoad', { caseLoadId: firstCaseLoad })
-      return {
-        ...details,
-        activeCaseLoadId: firstCaseLoad,
-      }
+const userServiceFactory = (elite2Api, oauthApi) => {
+  async function getActiveCaseloadAndSetIfNotSet(context, details) {
+    const caseloads = await elite2Api.getCaseLoads(context)
+
+    const activeCaseLoad = caseloads.find(cl => cl.currentlyActive)
```

Comment:
This seems to be duplicated in various places. In some cases, we're checking for nulls and others were not. Do you think it's worth stuffing a getActiveCaseLoad in the elite2Api js ?